### PR TITLE
improve coverage for syncer/sharding_group.go to 82.4%

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
part of #998

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: syncer/sharding_group.go [hunk 1]
```diff
@@ -557,9 +542,11 @@ func (k *ShardingGroupKeeper) UnresolvedTables() (map[string]bool, [][]string) {
 	tables := make([][]string, 0, 10)
 	k.RLock()
 	defer k.RUnlock()
+	// TODO: where does k.groups delete its key?
```
Comment (csuzhangxc) [inline syncer/sharding_group.go:545]: Can we delete the key after all tables/schemas leave the group (the group becomes empty)?

**Review by csuzhangxc — COMMENTED** [review]
LGTM

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**